### PR TITLE
First approach for creating a proposal (ERC20Voting)

### PR DESCRIPTION
**Pull request by marcvelmer**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task: [ID]()

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: javascript/modules/client/src/client-dao.ts [hunk 1]
```diff
@@ -61,14 +65,33 @@ export class ClientDaoERC20Voting extends ClientCore
...
     simpleVote: {
-      createProposal: (
-        _startDate: number,
-        _endDate: number,
-        _executeApproved?: boolean,
-        _voteOnCreation?: boolean
-      ): Promise<string> => {
-        // TODO: Not implemented
-        return Promise.resolve("0x1234567890123456789012345678901234567890");
+      createProposal: async (
```
Comment (brickpop) [inline javascript/modules/client/src/client-dao.ts]: Why async if there is no await?

File: javascript/modules/client/src/internal/interfaces/dao.ts [hunk 2]
```diff
@@ -110,3 +108,25 @@ export interface VotingConfig {
...
+export interface ICreateProposal {
+  metadata: string;
+  actions?: ProposalAction[];
+  startDate?: number;
+  endDate?: number;
+  executeIfDecided?: boolean;
+  creatorChoice?: VoteOption;
+}
+
+export interface ProposalAction {
```
Comment (brickpop) [inline javascript/modules/client/src/internal/interfaces/dao.ts]:
```suggestion
export interface IProposalAction {
```

File: javascript/modules/client/src/internal/interfaces/dao.ts [hunk 3]
```diff
@@ -110,3 +108,25 @@ export interface VotingConfig {
...
+  to: string;
+  value: number;
+  data: string;
+}
+
+export enum VoteOption {
+  None,
+  Abstain,
+  Yea,
+  Nay,
```
Comment (brickpop) [inline javascript/modules/client/src/internal/interfaces/dao.ts]:
```suggestion
export enum VoteOptions {
  NONE,
  ABSTAIN,
  YEA,
  NAY,
```

File: javascript/modules/client/README.md [hunk 4]
```diff
@@ -67,6 +67,9 @@ const daoCreationParams: ICreateDaoERC20Voting = {
     gsnForwarder: Wallet.createRandom().address,
 };
 
+const gasFeesEstimation = await client.estimate.create(daoCreationParams);
+console.log(gasFeesEstimation) // Gas fee estimation
```
Comment (brickpop) [inline javascript/modules/client/README.md]: Could we illustrate an example of the printed value? to convey what the type is

File: javascript/modules/client/README.md [hunk 5]
```diff
@@ -102,10 +105,69 @@ const daoCreationParams: ICreateDaoWhitelistVoting = {
...
```
Comment (brickpop) [inline javascript/modules/client/README.md]: Same as above

File: javascript/modules/client/README.md [hunk 6]
```diff
@@ -102,10 +105,69 @@ const daoCreationParams: ICreateDaoWhitelistVoting = {
...
+    signer: new Wallet("privateKey"),
+    daoFactoryAddress: "daoFactoryAddress",
+    web3Providers: ["http://localhost:8545"],
+};
+const context = new Context(contextParams);
+
+const client = new ClientDaoERC20Voting(context);
+
+const proposalCreationParams: ICreateProposal = {
+    metadata: "0x1234",
```
Comment (brickpop) [inline javascript/modules/client/README.md]: What does this actually mean? An IPFS hash? Shouldn't it be the IPFS CID?

File: javascript/modules/client/src/client-dao.ts [hunk 7]
```diff
@@ -62,13 +65,32 @@ export class ClientDaoERC20Voting extends ClientCore
...
-        _voteOnCreation?: boolean
-      ): Promise<string> => {
-        // TODO: Not implemented
-        return Promise.resolve("0x1234567890123456789012345678901234567890");
+        votingAddress: string,
+        params: ICreateProposal
+      ): Promise<BigNumber> => {
+        if (!this.signer)
+          throw new Error("A signer is needed for creating a DAO");
+        if (!votingAddress)
```
Comment (brickpop) [inline javascript/modules/client/src/client-dao.ts]:
```suggestion
        else if (!votingAddress)
```

File: javascript/modules/client/src/client-dao.ts [hunk 8]
```diff
@@ -62,13 +65,32 @@ export class ClientDaoERC20Voting extends ClientCore
...
-        _executeApproved?: boolean,
-        _voteOnCreation?: boolean
-      ): Promise<string> => {
-        // TODO: Not implemented
-        return Promise.resolve("0x1234567890123456789012345678901234567890");
+        votingAddress: string,
+        params: ICreateProposal
+      ): Promise<BigNumber> => {
+        if (!this.signer)
+          throw new Error("A signer is needed for creating a DAO");
```
Comment (brickpop) [inline javascript/modules/client/src/client-dao.ts]:
```suggestion
          return Promise.reject(new Error("A signer is needed for creating a DAO"));
```

File: javascript/modules/client/src/client-dao.ts [hunk 9]
```diff
@@ -62,13 +65,32 @@ export class ClientDaoERC20Voting extends ClientCore
...
+        if (!votingAddress)
+          throw new Error(
+            "A voting contract address is needed for creating a proposal"
+          );
```
Comment (brickpop) [inline javascript/modules/client/src/client-dao.ts]:
```suggestion
          return Promise.reject(new Error(
            "A voting contract address is needed for creating a proposal"
          ));
```

File: javascript/modules/client/src/client-dao.ts [hunk 10]
```diff
@@ -185,13 +207,32 @@ export class ClientDaoWhitelistVoting extends ClientCore
...
-        _executeApproved?: boolean,
-        _voteOnCreation?: boolean
-      ): Promise<string> => {
-        // TODO: Not implemented
-        return Promise.resolve("0x1234567890123456789012345678901234567890");
+        votingAddress: string,
+        params: ICreateProposal
+      ): Promise<BigNumber> => {
+        if (!this.signer)
+          throw new Error("A signer is needed for creating a DAO");
```
Comment (brickpop) [inline javascript/modules/client/src/client-dao.ts]: Same as above

File: javascript/modules/client/test/unit/client.test.ts [hunk 11]
```diff
@@ -282,4 +283,42 @@ describe("Client instances", () => {
     expect(newDaoAddress).toContain("0x");
     expect(newDaoAddress).toMatch(/^[A-Fa-f0-9]/i);
   });
+  // it("Should create a ERC20Voting proposal locally", async () => {
```
Comment (brickpop) [inline javascript/modules/client/test/unit/client.test.ts:286]: Why these commented lines?

**Review by emmdim — APPROVED** [review]

**Review by brickpop — APPROVED** [review]

**Pull request merged**